libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -997,16 +997,64 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		break;
 	case FunctionType::Kind::Internal:
 	{
-		FunctionDefinition const* functionDef = ASTNode::resolveFunctionCall(_functionCall, &m_context.mostDerivedContract());
-
+		solAssert(functionType);
 		solAssert(!functionType->takesArbitraryParameters());
 
 		vector<string> args;
-		if (functionType->hasBoundFirstArgument())
-			args += IRVariable(_functionCall.expression()).part("self").stackSlots();
+		FunctionDefinition const* functionDef = nullptr;
 
-		for (size_t i = 0; i < arguments.size(); ++i)
-			args += convert(*arguments[i], *parameterTypes[i]).stackSlots();
+		if (!_functionCall.isSuffixCall())
```

Comment:
Can't the case distinction here be fully reduced to a ``if (_functionCall.isSuffixCall() && parameterTypes.size() == 2)`` and splitting the arguments in that case, while leaving the rest be handled like before?

Reply:
Done.